### PR TITLE
refactor(error handling crud)

### PR DESCRIPTION
**Pull request by ChrOertlin**
## Description

Adds raising a value error in the crud when retrieving entires by unique identifiers for illumina metrics and flow cells

### Added

-

### Changed

-

### Fixed

-

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/store/crud/create.py [hunk 1]
```diff
@@ -398,16 +398,18 @@ def link_case_to_order(order_id: int, case_id: int):
...
+        except ValueError:
+            new_flow_cell = IlluminaFlowCell(
+                internal_id=flow_cell_dto.internal_id,
+                type=flow_cell_dto.type,
+                model=flow_cell_dto.model,
+            )
+            self.session.add(new_flow_cell)
+            LOG.debug(f"Flow cell added to status db: {new_flow_cell.internal_id}.")
+            return new_flow_cell
+        raise ValueError(f"Flow cell already exists: {flow_cell_dto.internal_id}")
```
Comment (seallard) [inline cg/store/crud/create.py:412]: I think it is better to raise a custom exception here, like `FlowCellNotFound` or something similar

**Review by seallard — APPROVED** [review]
Nice! I think it makes sense in our context to treat a missing record in these methods as an exceptional case rather than having an optional return type. We always expect an entry to exist when we call these methods.

It would be nice to lift out the error handling into a decorator, but lets leave it for now 👍 It would be great to have more descriptive names for the exceptions than just reusing the value error

File: cg/store/crud/create.py (hunk 1 above)
Comment (ChrOertlin) [reply cg/store/crud/create.py]: I added an "EntryNotFoundError" and "EntryAlreadyExistsError" where applicable.

**Comment by ChrOertlin** [comment]
> Nice! I think it makes sense in our context to treat a missing record in these methods as an exceptional case rather than having an optional return type. We always expect an entry to exist when we call these methods.
> 
> It would be nice to lift out the error handling into a decorator, but lets leave it for now 👍 It would be great to have more descriptive names for the exceptions than just reusing the value error

I can add the decorators in another PR!

**Pull request merged**